platform/api/src/api/v1/gql/guards/mod.rs

```diff
@@ -0,0 +1,32 @@
+use async_graphql::Context;
+use ulid::Ulid;
+
+use crate::database::RolePermission;
+
+use super::{
+    error::{GqlError, Result},
+    ext::ContextExt,
+};
+
+// This can't be replaced by async_graphql's field guards because of this: https://github.com/async-graphql/async-graphql/issues/1398
+// I don't see a better alternative than doing this for now.
+pub async fn auth_guard<T>(
+    ctx: &Context<'_>,
+    field_name: &'static str,
+    field_value: T,
+    user_id: Ulid,
+) -> Result<T> {
```

Comment:
Might it be worth adding the ability to specify what permissions are required?

Reply:
Yeah, we could do that but it isn't needed right now

Reply:
Okay, seems good then.